resources/sqlite/src/conn_opts.rs

```diff
@@ -0,0 +1,215 @@
+use std::path::Path;
+use std::str::FromStr;
+use std::time::Duration;
+use std::{borrow::Cow, path::PathBuf};
+
+use serde::{Deserialize, Serialize};
+use sqlx::sqlite::SqliteConnectOptions;
+
+mod journal_mode;
+pub use journal_mode::*;
+
+mod synchronous;
+pub use synchronous::*;
+use tracing::debug;
+
+/// Options to configure the SQLite database mirroring sqlx's [`SqliteConnectOptions`](https://docs.rs/sqlx/latest/sqlx/sqlite/struct.SqliteConnectOptions.html)
+/// for the options it exposes, see their docs for reference.
+///
+/// Construction of the full connection string is handled internally for security reasons and defaults to creating a
+/// file-based database named `default_db.sqlite` with `create_if_missing == true`. Use the `filename` and/or
+/// `in_memory` methods to configure the type of database created.
+///
+/// Note that Shuttle does currently not support the `collation`, `thread_name`, `log_settings`, `pragma`, `extension`,
+/// `shared_cache` options. The following options are internally controlled by pragmas and hence also not exposed: `foreign_keys`, `locking_mode`, `auto_vacuum`, `page_size`.
+#[derive(Debug, Deserialize, Serialize)]
+pub struct ShuttleSqliteConnOpts {
```

Comment:
I had updated the names already but didn't push yet because I hadn't rebased onto the newest main yet. I opted for the longer but explicit `ShuttleSqlite` and `ShuttleSqliteConnOpts`, hope that's fine!